### PR TITLE
item resource tests

### PR DESCRIPTION
**Pull request by TahaKhanAbdalli**
Supports #180

### CODE REVIEW

File: src/providers/dataProvider/tests/item.test.ts [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+      prefixLocalForageKey: TEST_STORAGE_KEY
+    })
+    provider = withLifecycleCallbacks(
+      withOutLifecycleProvider,
+      lifecycleCallbacks(
+        trackEvent(withOutLifecycleProvider),
+        withOutLifecycleProvider
+      )
+    )
+
```
Comment (IanMayo) [inline src/providers/dataProvider/tests/item.test.ts:53]: @taha - there are two very similar blocks in the next few lines.

I think we should declare an array of resource-names, then loop through the names clearing each.  We should use the same strategy in batch-test.

We will have tests for other resource types in the future, so we will just end up passing in the full list of resource names.

Comment (taha) [reply src/providers/dataProvider/tests/item.test.ts]: You have the wrong Taha there 😅

File: src/providers/dataProvider/tests/batch.test.ts [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+      pagination: { page: 1, perPage: 1000 },
+      filter: {}
+    })
+    if (list.total !== undefined && list.total > 0) {
+      await provider.deleteMany(R_BATCHES, {
+        ids: list.data.map((item) => item.id)
+      })
+    }
+
+    const auditList = await provider.getList<Audit>(R_AUDIT, {
```
Comment (IanMayo) [inline src/providers/dataProvider/tests/batch.test.ts:57]: @TahaKhanAbdalli - we should be using your generic clear method here in this class.

File: src/providers/dataProvider/index.ts [hunk 3]
```diff
@@ -216,7 +216,7 @@ const auditForUpdatedChanges = async (
     activityDetail: `Previous values: ${JSON.stringify(difference)}`,
     resource: constants.R_ITEMS,
     index: record.id as number,
```
Comment (IanMayo) [inline src/providers/dataProvider/index.ts:218]: @TahaKhanAbdalli  - I don't see why we are using `previousData` in the next line.  Are the two ids always the same?

Comment (TahaKhanAbdalli) [reply src/providers/dataProvider/index.ts:218]: Yes the `ids` are same, when updating `react-admin` provides the full current document to update in `record.data` but when we are manually updating in the test we provided `partial` of the data to update which was causing issues in testing. Or I can remove this and provide the id when updating the document then we can use `record.data.id`

Comment (IanMayo) [reply src/providers/dataProvider/index.ts:218]: Aah, I see.  That's ok :-)

**Review by IanMayo — APPROVED** [review]

**Pull request merged**